### PR TITLE
dai: simple i/f to control flow between platform and lib

### PR DESCRIPTION
**Pull request by mmaka1**
A mechanism similar to the one used for dma applied.
Dai initialization moved to separate unit from the main platform
code for cavs.
Optional dynamic discovery and initialization possible before
making available to other fw units via the lib hub.
SSP count cleaned up for cavs (important pre-work
for clock gating).

Signed-off-by: Marcin Maka <marcin.maka@linux.intel.com>

### CODE REVIEW

**Comment by mmaka1** [comment]
@lgirdwood CI seems to suffer from random t/o between the nodes. Let's wait till Mon, I'd prefer to go through the regression tests first.

File: src/lib/dai.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	lib_dai.dai_type_array = dai_type_array;
+	lib_dai.num_dai_types = num_dai_types;
+}
+
+struct dai *dai_get(uint32_t type, uint32_t index)
+{
+	int i;
+	struct dai_type_info *dti;
+
+	for (dti = lib_dai.dai_type_array;
```
Comment (lgirdwood) [inline src/lib/dai.c]: just noticed this, we will need to use {} around and inside this loop due to multiple loops and conditions - not sure why checkpatch did not complain though.

File: src/platform/cannonlake/include/platform/dai.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
```
Comment (lgirdwood) [inline src/platform/cannonlake/include/platform/dai.h]: oh, you should really be the author of any new files.

**Comment by mmaka1** [comment]
@lgirdwood Applied {} and fixed auth.

**Comment by lgirdwood** [comment]
@mmaka1 apologies, we have a conflict as a result an earlier merge.

**Comment by mmaka1** [comment]
@lgirdwood I am extending the patch to include Sue config for cavs family  too.

**Pull request merged**